projects/RabbitMQ.Client/RabbitMQ.Client.csproj

```diff
@@ -5,8 +5,6 @@
     <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings>
     <NoWarn>$(NoWarn);CS1591</NoWarn>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <EnableTrimAnalyzer Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">true</EnableTrimAnalyzer>
```

Comment:
Since this library doesn't target net6.0+, these properties don't have any benefit.

Since we can't use the analyzers, to ensure the library doesn't have any warnings, I followed the steps outlined in https://devblogs.microsoft.com/dotnet/creating-aot-compatible-libraries/#publishing-a-test-application-for-aot.

Let me know if you want that app in the repo. I can add it to this PR.

Reply:
@eerhardt having that test app would be great. I can add it to the github workflow.

Sorry it has taken me so long to reply.

Reply:
Do you want it in this PR or in a separate one?

Reply:
This one would be great! I've been so caught up in version 7 I've been behind on 6.9.0

Reply:
Ok, I've added the project in the latest commit. It can be `dotnet publish`'d and then the output inspected for warnings like in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/build/test-aot-compatibility.ps1.

Note there is no use in running the published app, like that example does. In their repo, they have tests they want executed, but here we just need the output of publish.

Reply:
Note in order to make this work with the .NET 6 SDK, you need to pass in a `--runtime` when you `dotnet publish`. So you can say `dotnet publish --runtime win-x64`.

For newer SDKs, this isn't needed because it uses the runtime of the current machine.

Reply:
Thank you for all of the information!